lib/api/v3/work_packages/work_package_list_helpers.rb

```diff
@@ -0,0 +1,165 @@
+          apply_sorting query, query_params
+          groups = apply_and_generate_groups query, query_params
+
+          total_sums = generate_total_sums query.results, query_params
+
+          collection_representer(query.results.sorted_work_packages,
+                                 project: project,
+                                 query_params: query_params,
+                                 groups: groups,
+                                 sums: total_sums)
+        end
+
+        def apply_filters(query, query_params)
+          if params[:filters]
+            set_filters_from_json(query, params[:filters])
+            query_params[:filters] = params[:filters]
+          end
+        end
+
+        def set_filters_from_json(query, json)
+          filters = JSON.parse(json)
+          operators = {}
+          values = {}
+          filters.each do |filter|
+            attribute = filter.keys.first # there should only be one attribute per filter
+            operators[attribute] = filter[attribute]['operator']
+            values[attribute] = filter[attribute]['values']
+          end
+
+          query.filters = []
+          query.add_filters(filters.map(&:keys).flatten, operators, values)
+        end
+
+        def apply_sorting(query, query_params)
+          if params[:sortBy]
+            set_sorting_from_json(query, params[:sortBy])
+            query_params[:sortBy] = params[:sortBy]
+          end
+        end
+
+        def set_sorting_from_json(query, json)
+          query.sort_criteria = JSON.parse(json)
+        end
+
+        def apply_and_generate_groups(query, query_params)
+          if params[:groupBy]
+            query.group_by = params[:groupBy]
+            query_params[:groupBy] = params[:groupBy]
+
+            generate_groups query.results
+          end
+        end
+
+        def generate_groups(results)
+          results.work_package_count_by_group.map { |group, count|
+            group_element = {
+              value: group.to_s,
+              count: count
+            }
+
+            # TODO: insert valueLink
+            if params[:showSums] == 'true'
+              group_element[:sums] = format_query_sums results.all_sums_for_group(group)
+            end
+
+            group_element
+          }
+        end
+
+        def generate_total_sums(results, query_params)
+          if params[:showSums] == 'true'
+            query_params[:showSums] = 'true'
+            format_query_sums results.all_total_sums
+          end
+        end
+
+        def format_query_sums(sums)
+          sums = format_column_keys sums
+          format_durations! sums
+        end
+
+        def format_column_keys(hash_by_column)
+          converter = API::Utilities::PropertyNameConverter
+          ::Hash[hash_by_column.map { |column, value|
+                   column_name = converter.from_ar_name(column.name.to_s)
+                   [column_name, value]
+                 }]
```

Comment:
`end` at 126, 17 is not aligned with `::Hash[hash_by_column.map { |column, value|` at 123, 10